sdks/java/core/src/main/java/org/apache/beam/sdk/util/construction/DataflowGroupByKeyTranslation.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.util.construction;
+
+import com.google.auto.service.AutoService;
+import java.util.Collections;
+import java.util.Map;
+import org.apache.beam.model.pipeline.v1.RunnerApi;
+import org.apache.beam.model.pipeline.v1.RunnerApi.FunctionSpec;
+import org.apache.beam.sdk.runners.AppliedPTransform;
+import org.apache.beam.sdk.transforms.DataflowGroupByKey;
+import org.apache.beam.sdk.transforms.PTransform;
+
+/**
+ * Utility methods for translating a {@link DataflowGroupByKey} to and from {@link RunnerApi}
```

Comment:
Is this necessary? Maybe you need it in order to get the URN in some context? Because otherwise obvs it cannot really go to/from runner API since the translator does not preserve all the information. To be clear: I feel like we shouldn't have this file, unless we need it as a technicality.

Reply:
Yes I think it is needed otherwise I got the following error:
```
Exception in thread "main" java.lang.IllegalArgumentException: Transform Redistribute-Redistribute-RedistributeByKey-DataflowGroupByKey is not a composite transform but does not have a specified URN. inputs {
  key: "org.apache.beam.sdk.values.PCollection.<init>:397#c574c40af2bc3c2d"
  value: "Redistribute/Redistribute.RedistributeByKey/ReifyOriginalMetadata/ParDo(Anonymous)/ParMultiDo(Anonymous).output"
}
outputs {
  key: "org.apache.beam.sdk.values.PCollection.<init>:397#b672aefab09fd4e6"
  value: "Redistribute/Redistribute.RedistributeByKey/DataflowGroupByKey.out"
}
unique_name: "Redistribute/Redistribute.RedistributeByKey/DataflowGroupByKey"
```

Reply:
Actually, I am able to delete this file and just add the contents to DataflowGBK.